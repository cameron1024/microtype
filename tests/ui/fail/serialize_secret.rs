microtype_macro::microtype! {
    secret String {
        Password
    }
}


fn main() {
    use microtype::SecretMicrotype;
    let password = Password::new("asdf".into());
    assert_serialize(password);
}

fn assert_serialize<T: serde::Serialize>(_t: T) {}
