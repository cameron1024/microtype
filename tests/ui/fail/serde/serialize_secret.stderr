error[E0277]: the trait bound `Password: Serialize` is not satisfied
  --> tests/ui/fail/serde/serialize_secret.rs:12:22
   |
12 |     assert_serialize(password);
   |     ---------------- ^^^^^^^^ the trait `Serialize` is not implemented for `Password`
   |     |
   |     required by a bound introduced by this call
   |
   = help: the following other types implement trait `Serialize`:
             &'a T
             &'a mut T
             ()
             (T0, T1)
             (T0, T1, T2)
             (T0, T1, T2, T3)
             (T0, T1, T2, T3, T4)
             (T0, T1, T2, T3, T4, T5)
           and 125 others
note: required by a bound in `assert_serialize`
  --> tests/ui/fail/serde/serialize_secret.rs:15:24
   |
15 | fn assert_serialize<T: serde::Serialize>(_t: T) {}
   |                        ^^^^^^^^^^^^^^^^ required by this bound in `assert_serialize`
